### PR TITLE
docs(auth): add note about using user token vs uid

### PR DESCRIPTION
**Pull request by aruhn1**
Added information about the ID Token which should be used to authenticate with the backend server.

### Checklist

- I read the [Contributor Guide](../CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
  - [ ] Yes
- My change supports the following platforms;
  - [ ] `Android`
  - [ ] `iOS`
- My change includes tests;
  - [ ] `e2e` tests added or updated in `packages/\*\*/e2e`
  - [ ] `jest` tests added or updated in `packages/\*\*/__tests__`
- [ ] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - [ ] Yes
  - [ ] No

### Test Plan

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Star this repo on GitHub ⭐️
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter

### CODE REVIEW

File: docs/auth/usage/index.md [hunk 1]
```diff
@@ -170,6 +170,19 @@ with the [`User`](/reference/auth/user) details.
...
+```
+     import auth from '@react-native-firebase/auth';
+     const user = auth().currentUser;
+     user
+        .getIdToken()
+        .then(token =>
+          console.log('uid token is ', token),
+        ); 
+```
+
```
Comment (mikehardy) [inline docs/auth/usage/index.md:184]:
```suggestion
The user's token should be used for authentication with your backend systems. The token is fetched with the [getIdToken](https://rnfirebase.io/reference/auth/user#getIdToken) method as shown in the following snippet. As mentioned in the [Firebase documentation](https://firebase.google.com/docs/auth/web/manage-users#get_a_users_profile), the uid should not be used for authentication. 
```
     import auth from '@react-native-firebase/auth';
     const user = auth().currentUser;
     user
        .getIdToken()
        .then(token =>
          console.log('user token is ', token),
        ); 
```
```

**Review by mikehardy — CHANGES_REQUESTED** [review]
I like this one also, some small wording change but basically good to go, what do you think?

**Comment by aruhn1** [comment]
> I like this one also, some small wording change but basically good to go, what do you think?

Yeah, it's better.

**Review by mikehardy — APPROVED** [review]

File: docs/auth/usage/index.md [hunk 2]
```diff
@@ -170,6 +170,9 @@ with the [`User`](/reference/auth/user) details.
 In case any errors occur, the module provides support for identifying what specifically went wrong by attaching a code
 to the error. For a full list of error codes available, view the [Firebase documentation](https://firebase.google.com/docs/reference/js/firebase.auth.Auth#error-codes_3).
 
+## Authenticate with backend server
+The user's token should be used for authentication with your backend systems. The token is fetched with the [getIdToken](https://rnfirebase.io/reference/auth/user#getIdToken) method. As mentioned in the [Firebase documentation](https://firebase.google.com/docs/auth/web/manage-users#get_a_users_profile), the uid should not be used for authentication.
```
Comment (mikehardy) [inline docs/auth/usage/index.md:174]: I believe prettier requires a blank line here to pass lint check, going to try it

```suggestion
## Authenticate with backend server

The user's token should be used for authentication with your backend systems. The token is fetched with the [getIdToken](https://rnfirebase.io/reference/auth/user#getIdToken) method. As mentioned in the [Firebase documentation](https://firebase.google.com/docs/auth/web/manage-users#get_a_users_profile), the uid should not be used for authentication. 
```

**Comment by mikehardy** [comment]
What a little pile of extra things to do :laughing:  - added two new non-english words (getIdToken / uid) to the spellcheck dictionary so that would pass, fixed up the formatting (my own code suggestion had an incorrect extra space).

CI should pass now then I can merge this thing

**Pull request merged**